### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by andrewshkovskii**

### CODE REVIEW

File: instagram/__init__.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
-from bind import InstagramAPIError, InstagramClientError
-from client import InstagramAPI
+from .bind import InstagramAPIError, InstagramClientError
+from .client import InstagramAPI
```
Comment (homm) [inline instagram/__init__.py]: Add `from __future__ import absolute_import` everywhere

File: instagram/bind.py [hunk 2]
```diff
@@ -67,7 +70,7 @@ def _build_parameters(self, args, kwargs):
                 except IndexError:
                     raise InstagramClientError("Too many arguments supplied")
 
-            for key, value in kwargs.iteritems():
+            for key, value in six.iteritems(kwargs):
```
Comment (homm) [inline instagram/bind.py]: `kwargs.items()`

File: instagram/bind.py [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+from six.moves.urllib.parse import quote
+
 
 re_path_template = re.compile('{\w+}')
 
 
 def encode_string(value):
     return value.encode('utf-8') \
-                        if isinstance(value, unicode) else str(value)
+        if isinstance(value, six.text_type) else str(value)
```
Comment (homm) [inline instagram/bind.py]: `str(value)` will give you: 1. unicode type 2. representation of the original string:

```python
>>> str(b'pr\xc3\xb3ba')
"b'pr\\xc3\\xb3ba'"
```

File: instagram/models.py [hunk 4]
```diff
@@ -10,7 +12,13 @@ def object_from_dictionary(cls, entry):
         return cls(**entry_str_dict)
 
     def __repr__(self):
-        return unicode(self).encode('utf8')
+        return str(self)
+
+    def __str__(self):
+        if six.PY3:
+            return self.__unicode__()
```
Comment (homm) [inline instagram/models.py]: There is no `__unicode__` method.

File: instagram/models.py [hunk 5]
```diff
@@ -34,7 +42,7 @@ class Media(ApiModel):
 
     def __init__(self, id=None, **kwargs):
         self.id = id
-        for key, value in kwargs.iteritems():
+        for key, value in six.iteritems(kwargs):
```
Comment (homm) [inline instagram/models.py]: `kwargs.items()` everywhere

File: instagram/oauth2.py [hunk 6]
```diff
@@ -106,7 +105,7 @@ def exchange_for_access_token(self, code=None, username=None, password=None, sco
         http_object = Http(disable_ssl_certificate_validation=True)
         url = self.api.access_token_url
         response, content = http_object.request(url, method="POST", body=data)
-        parsed_content = simplejson.loads(content)
+        parsed_content = simplejson.loads(content.decode())
```
Comment (homm) [inline instagram/oauth2.py]: specify 'utf-8'

**Comment by andrewshkovskii** [comment]
@homm Thanks for the feedback. 
This was just patches from another repo - just copy and paste without any doubt.
I will fix all issues :)

File: instagram/models.py (hunk 4 above)
Comment (andrewshkovskii) [reply instagram/models.py]: @homm Every class in this file has it, it was there before these changes.

File: instagram/bind.py (hunk 3 above)
Comment (andrewshkovskii) [reply instagram/bind.py]: @homm Did you mean that `str` call is redundant?
If so - this code should work on py2 and py3, so your example for py2 will have different result

```
Python 2.7.10 (default, Jul 14 2015, 19:46:27)
[GCC 4.8.2] on linux
> str(b'pr\xc3\xb3ba')
'pr\xc3\xb3ba'
```

Comment (homm) [reply instagram/bind.py]: > Python 2.7.10

???

Comment (andrewshkovskii) [reply instagram/bind.py]: @homm Clarify question, please.

File: instagram/models.py [hunk 7]
```diff
@@ -9,8 +13,17 @@ def object_from_dictionary(cls, entry):
         entry_str_dict = dict([(str(key), value) for key, value in entry.items()])
```
Comment (homm) [inline instagram/models.py]: `str(key)` could be cause of error

**Comment by andrewshkovskii** [comment]
@homm Please, merge. I can't merge this PR.

**Pull request merged**